### PR TITLE
py-ansible,py-ansible-core,py-resolvelib update to latest versions

### PR DESCRIPTION
**Pull request by jhoyt4**
py-ansible
  - update to version 9.2.0
  - add python312 compatibility 
  
  py-ansible-core
  - update to version  2.16.3
  - add python312 compatibility
  
  py-resolvelib
  - update to version 1.0.1
  - add python312 compatibility

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
macOS 14.2.1 23C71 x86_64
Xcode 15.2 15C500b

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by jhoyt4** [comment]
Testing is underway, will convert back from draft when testing is complete

**Comment by jhoyt4** [comment]
tested, through checks, and ready for review

**Review by steenzout — APPROVED** [review]
LGTM

File: python/py-resolvelib/Portfile [hunk 1]
```diff
@@ -17,16 +17,16 @@ maintainers         nomaintainer
...
+                    sha256  04ce76cbd63fded2078ce224785da6ecd42b9564b1390793f64ddecbe997b309 \
+                    size    21065
 
 long_description \
     ResolveLib at the highest level provides a Resolver class that includes dependency resolution logic. \
     You give it some things, and a little information on how it should interact with them, and it will \
     spit out a resolution result.
 
-python.versions     37 38 39 310 311
+python.versions     37 38 39 310 311 312
```
Comment (pmetzger) [inline python/py-resolvelib/Portfile:29]: I am going to commit, but it would be useful if you submitted another PR some time soon removing the obsolete 3.7 Python version if that is possible.

Comment (jhoyt4) [reply python/py-resolvelib/Portfile:29]: done - https://github.com/macports/macports-ports/pull/22550
